scipy/stats/tests/test_continuous.py

```diff
@@ -1148,6 +1148,80 @@ def pdf(self, x, a, b):
                 assert_allclose(X.moment(order, kind=kind),
                                 Y.moment(order, kind=kind))
 
+    # pdf and cdf formulas below can warn on boundary of support in some cases.
+    # See https://github.com/scipy/scipy/pull/22560#discussion_r1962763840.
+    @pytest.mark.filterwarnings("ignore::RuntimeWarning")
+    @pytest.mark.parametrize("c", [-1, 0, 1, np.asarray([-2.1, -1., 0., 1., 2.1])])
+    def test_custom_variable_support(self, c):
+        rng = np.random.default_rng(7548723590230982)
+
+        class MyGenExtreme:
+            @property
+            def __make_distribution_version__(self):
+                return "1.16.0"
+
+            @property
+            def parameters(self):
+                return {'c': {'endpoints': (-np.inf, np.inf), 'inclusive': (False, False)},
+                        'mu': {'endpoints': (-np.inf, np.inf), 'inclusive': (False, False)},
+                        'sigma': {'endpoints': (0, np.inf), 'inclusive': (False, False)}}
+
+            @property
+            def support(self):
+                def left(*, c, mu, sigma):
+                    c, mu, sigma = np.broadcast_arrays(c, mu, sigma)
+                    result = np.empty_like(c)
+                    result[c >= 0] = -np.inf
+                    result[c < 0] = mu[c < 0] + sigma[c < 0] / c[c < 0]
+                    return result[()]
+
+                def right(*, c, mu, sigma):
+                    c, mu, sigma = np.broadcast_arrays(c, mu, sigma)
+                    result = np.empty_like(c)
+                    result[c <= 0] = np.inf
+                    result[c > 0] = mu[c > 0] + sigma[c > 0] / c[c > 0]
+                    return result[()]
+
+                return {"endpoints": (left, right), "inclusive": (False, False)}
+
+            def pdf(self, x, *, c, mu, sigma):
+                x, c, mu, sigma = np.broadcast_arrays(x, c, mu, sigma)
+                t = np.empty_like(x)
+                mask = (c == 0)
+                t[mask] = np.exp(-(x[mask] - mu[mask])/sigma[mask])
+                t[~mask] = (1  - c[~mask]*(x[~mask] - mu[~mask])/sigma[~mask])**(1/c[~mask])
```

Comment:
Not needed now because `_lazywhere` is being transitioned to `xpx.apply_where`, but just to show what it would look like:
```suggestion
                t = _lazywhere(c == 0, (x, c, mu, sigma),
                               lambda x, c, mu, sigma: np.exp(-(x - mu)/sigma),
                               lambda x, c, mu, sigma: (1 - c*(x - mu)/sigma)**(1 / c))
```
A bit easier on the eyes, and I think it does the broadcasting for you.
But maybe it's because I was looking at `apply_where` PR for an hour and got used to it.

Reply:
I like `_lazywhere` and have used it before. I was thinking since this code  is supposed to be an example of a user created custom distribution, it wouldn't be good to rely on something outside SciPy's public API. I was thinking it might be good to put this example in the documentation.

Reply:
Mmm. yes, I was thinking we need an example of arbitrary function `support` in he documentation, too, and you're right that we wouldn't want to use a private function there.